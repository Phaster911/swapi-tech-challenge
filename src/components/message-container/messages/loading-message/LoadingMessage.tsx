import styled from "styled-components";
import { COLORS } from "../../../../style-constants";
import StyledMessageSection from "../../../shared/styled-message-wrapper/StyledMessageWrapper";

const svgString = `
  <svg viewBox='-33 -33 66 66' xmlns='http://www.w3.org/2000/svg'>
  <defs>
    <linearGradient id='fadeGradient' x1='0%' y1='0%' x2='100%' y2='0%'>
      <stop offset='0%' style='stop-color:${COLORS.accent}; stop-opacity:1' />
      <stop offset='100%' style='stop-color:${COLORS.accent}; stop-opacity:0' />
    </linearGradient>
  </defs>
  <circle fill='none' stroke-width='6' stroke-dasharray='200' stroke='url(#fadeGradient)' stroke-dashoffset='56' r='30'/>
</svg>
`;

const encodedSvg = encodeURIComponent(svgString.trim());

const backgroundSvg = `url("data:image/svg+xml,${encodedSvg}")`;

const StyledProgress = styled.progress`
  appearance: none;
  display: block;
  background: none;
  width: 1em;
  height: 1em;
  border: 0;

  &::-moz-progress-bar {
    background: none;
  }
  &::-webkit-progress-bar {
    display: none;
  }

  background-image: ${backgroundSvg};

  width: 6em;
  height: 6em;
  animation: spin 1.5s linear infinite;

  @keyframes spin {
    to {
      transform: rotate(360deg);
    }
  }
`;

export default function LoadingMessage() {
  return (
    <StyledMessageSection
      aria-labelledby="loading-message-title"
      role="status"
      aria-live="assertive"
      data-testid="loading-message-testid"
    >
      <h2 id="loading-message-title">Loading...</h2>
      <label>
        <StyledProgress aria-label="Please wait while we fetch your search results." />
      </label>
    </StyledMessageSection>
  );
}
