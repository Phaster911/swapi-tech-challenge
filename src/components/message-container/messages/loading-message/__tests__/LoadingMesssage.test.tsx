import { render } from "@testing-library/react";
import LoadingMessage from "../LoadingMessage";

describe("<LoadingMessage />", () => {
  it("renders the component", () => {
    const { container } = render(<LoadingMessage />);
    expect(container).toMatchSnapshot();
  });
});
