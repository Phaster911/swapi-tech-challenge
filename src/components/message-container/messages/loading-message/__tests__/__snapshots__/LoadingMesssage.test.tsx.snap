// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LoadingMessage /> > renders the component 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1rem;
  text-align: center;
}

.c1 {
  appearance: none;
  display: block;
  background: none;
  width: 1em;
  height: 1em;
  border: 0;
  background-image: url("data:image/svg+xml,%3Csvg%20viewBox%3D'-33%20-33%2066%2066'%20xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg'%3E%0A%20%20%3Cdefs%3E%0A%20%20%20%20%3ClinearGradient%20id%3D'fadeGradient'%20x1%3D'0%25'%20y1%3D'0%25'%20x2%3D'100%25'%20y2%3D'0%25'%3E%0A%20%20%20%20%20%20%3Cstop%20offset%3D'0%25'%20style%3D'stop-color%3A%235da647%3B%20stop-opacity%3A1'%20%2F%3E%0A%20%20%20%20%20%20%3Cstop%20offset%3D'100%25'%20style%3D'stop-color%3A%235da647%3B%20stop-opacity%3A0'%20%2F%3E%0A%20%20%20%20%3C%2FlinearGradient%3E%0A%20%20%3C%2Fdefs%3E%0A%20%20%3Ccircle%20fill%3D'none'%20stroke-width%3D'6'%20stroke-dasharray%3D'200'%20stroke%3D'url(%23fadeGradient)'%20stroke-dashoffset%3D'56'%20r%3D'30'%2F%3E%0A%3C%2Fsvg%3E");
  width: 6em;
  height: 6em;
  animation: spin 1.5s linear infinite;
}

.c1::-moz-progress-bar {
  background: none;
}

.c1::-webkit-progress-bar {
  display: none;
}

<div>
  <div
    aria-labelledby="loading-message-title"
    aria-live="assertive"
    class="c0"
    data-testid="loading-message-testid"
    role="status"
  >
    <h2
      id="loading-message-title"
    >
      Loading...
    </h2>
    <label>
      <progress
        aria-label="Please wait while we fetch your search results."
        class="c1"
      />
    </label>
  </div>
</div>
`;
